processing/src/main/java/org/apache/druid/segment/ColumnarValueUnnestCursor.java

```diff
@@ -69,7 +94,11 @@ public ColumnSelectorFactory getColumnSelectorFactory()
       @Override
       public DimensionSelector makeDimensionSelector(DimensionSpec dimensionSpec)
       {
-        throw new UnsupportedOperationException("Dimension selector not applicable for column value selector");
+        if (!outputName.equals(dimensionSpec.getDimension())) {
+          return baseColumSelectorFactory.makeDimensionSelector(dimensionSpec);
+        }
+        throw new UnsupportedOperationException(
+            "Dimension selector not applicable for column value selector for column " + outputName);
```

Comment:
More design nits:

1. We have a `UOE` that bundles `String.format` into the building of the exception use that.
2. We encase interpolated values in `[]` to help differentiate things like extra spaces.